### PR TITLE
[Kiosk SDK] Adds support for attaching rules to Transfer Policy

### PR DESCRIPTION
**Pull request by manolisliolios**
## Description 

This PR adds support to attaching rules (royalty + kiosk lock) on a Transfer Policy. 
It also extends the README to add some more examples on how to use these.

## Test Plan 

How did you test the new or updated feature?

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: sdk/kiosk/README.md [hunk 1]
```diff
@@ -19,6 +18,27 @@ You can also use your preferred package manager, such as yarn or pnpm.
 
 Here are some indicative examples on how to use the kiosk SDK.
 
+<details>
+<summary>Find an addresses' owned kiosk(s)</summary>
```
Comment (randall-Mysten) [inline sdk/kiosk/README.md:22]: "Find an address" or "find addresses"
But maybe more clear to say "Find the kiosks owned by an address" or "Find the kiosks an address owns"

File: sdk/kiosk/README.md [hunk 2]
```diff
@@ -19,6 +18,27 @@ You can also use your preferred package manager, such as yarn or pnpm.
...
+	const address = `0xAddress`;
+	const { data } = await getOwnedKiosks(provider, address);
+	console.log(data); // kioskOwnerCaps:[], kioskIds: []
+};
+```
+
+</details>
+
 <details>
 <summary>Getting the listings & items by the kiosk's id</summary>
```
Comment (randall-Mysten) [inline sdk/kiosk/README.md:43]: just kiosk ID or use the label used in code and format as code inline

File: sdk/kiosk/README.md [hunk 3]
```diff
@@ -182,3 +202,77 @@ const withdraw = async () => {
...
+
+<summary>Create a Transfer Policy for a Type</summary>
+
+You can only create a TransferPolicy for packages that you own the `publisher` Object.
+
+It's recommended (unless you have a more advanced use case) to create only one transfer policy per type, or make sure that all transfer policies have the same rules.
+
+If you have multiple transfer policies, someone could purchase an item of that Type from a kiosk by using the easiest (to resolve) rules.
+
+You could extend the following snippet to do a check before creating the transfer policy, by using the `queryTransferPolicy` function, similar to the `purchaseAndResolvePolicies` example.
```
Comment (randall-Mysten) [inline sdk/kiosk/README.md:216]: As a best practice, you should use a single Transfer policy per type (T). If you use more than one policy for a type, and the rules for each differ, anyone that meets the conditions for any of the attached policies can purchase an asset from a kiosk. You can't specify which policy applies to a specific asset for the type when there is more than one policy attached. When someone meets the conditions that are easiest to meet, they are allowed to purchase and transfer the asset. 

Before you create a transfer policy, you can use the `quesryTransferpolicy` function to check the transfer policy associated with a type. This is similar to the `purchaseAndResolvePolicies` example.

Where is this example?

File: sdk/kiosk/README.md [hunk 4]
```diff
@@ -182,3 +202,77 @@ const withdraw = async () => {
...
+	// ... continue to sign and execute the transaction
+	// ...
+};
+```
+
+</details>
+
+<details>
+
+<summary>Attach Rules (Royalty, Kiosk Lock) to the Transfer Policy</summary>
```
Comment (randall-Mysten) [inline sdk/kiosk/README.md:242]:
```suggestion
<summary>Attach Royalty and Lock rules to a Transfer policy</summary>
```

**Review by damirka — APPROVED** [review]
Looks good but we need a note on basis points and how they're passed into the function. Perhaps, as "friendly" thing we could suggest formats? Like "0.0"? Or "0.10" or "98.10"?

**Pull request merged**